Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp

```diff
@@ -66,32 +66,38 @@ ViewGestureGeometryCollector::ViewGestureGeometryCollector(WebPage& webPage)
     , m_renderTreeSizeNotificationThreshold(0)
 #endif
 {
-    WebProcess::singleton().addMessageReceiver(Messages::ViewGestureGeometryCollector::messageReceiverName(), m_webPage->identifier(), *this);
+    WebProcess::singleton().addMessageReceiver(Messages::ViewGestureGeometryCollector::messageReceiverName(), webPage.identifier(), *this);
 }
 
 ViewGestureGeometryCollector::~ViewGestureGeometryCollector()
 {
+    RefPtr webPage = m_webPage.get();
+    if (!webPage)
+        return;
```

Comment:
This is concerning. This means we'll no longer remove ourselves from the WebProcess as a message receiver if the page is already gone.

Reply:
That's a good point. Given that we need the page identifier to remove ourselves, I just added a member variable to store the identifier so we won't need the page to access it.